cloudformation/ec2.yaml

```diff
@@ -0,0 +1,335 @@
+                  Restart=always
+                  [Install]
+                  WantedBy=multi-user.target
+          commands:
+            01_enable_cfn_hup:
+              command: !Sub |
+                systemctl enable cfn-hup.service
+            02_start_cfn_hup:
+              command: !Sub |
+                systemctl start cfn-hup.service
+    Properties:
+      ImageId: !FindInMap
+        - AWSRegion2AMI
+        - !Ref 'AWS::Region'
+        - AMI
+      InstanceType: !Ref 'InstanceType'
+      IamInstanceProfile: !Ref CloudwatchInstanceProfile
+      SecurityGroupIds:
+        - !Ref 'TorcheServeSecurityGroup'
+      SubnetId: !Ref 'SubnetId'
+      KeyName: !Ref 'KeyName'
+      UserData:
+        Fn::Base64: !Sub |
+          #!/bin/bash
+          wget https://s3.amazonaws.com/amazoncloudwatch-agent/ubuntu/amd64/latest/amazon-cloudwatch-agent.deb -O /tmp/amazon-cloudwatch-agent.deb
+          dpkg -i /tmp/amazon-cloudwatch-agent.deb
+          apt-get update -y
+          apt-get install -y python-pip python-setuptools openjdk-11-jdk openssl
+          python /usr/lib/python2.7/dist-packages/easy_install.py --script-dir /opt/aws/bin  https://s3.amazonaws.com/cloudformation-examples/aws-cfn-bootstrap-latest.tar.gz
+          # Install miniconda https://docs.anaconda.com/anaconda/install/silent-mode/
+          wget -q https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O /tmp/miniconda.sh
+          rm -rf /home/ubuntu/miniconda && bash /tmp/miniconda.sh -b -p /home/ubuntu/miniconda
+          export PATH="/home/ubuntu/miniconda/bin:$PATH"
+          conda init
+          source /home/ubuntu/miniconda/etc/profile.d/conda.sh
+          conda config --set always_yes yes
+          conda create --name torchserve torchserve torch-model-archiver psutil future pytorch torchtext torchvision cudatoolkit=10.1 -c pytorch -c powerai
+          conda activate torchserve
+          # Setup private-key file and self-signed certificate files
+          mkdir /etc/torchserve/
+          openssl req -new -x509 -days 365 -nodes -out /etc/torchserve/server.pem -keyout /etc/torchserve/server.key -subj "/C=US/ST=Washington/L=Seattle/O=Amazon Web Services/OU=Torchserve"
+          # Start cfn-hup
+          /opt/aws/bin/cfn-init -v --stack ${AWS::StackId} --resource TorchServeInstance --region ${AWS::Region} --configsets InstallAndRun
+          /opt/aws/bin/cfn-signal -e $? --stack ${AWS::StackId} --resource TorchServeInstance --region ${AWS::Region}
+          # Start torchserve
+          mkdir model_store
+          export LOG_LOCATION="/var/log/torchserve"
+          torchserve --start --model-store ./model_store --ts-config /etc/torchserve/config.properties
+    CreationPolicy:
+      ResourceSignal:
+        Timeout: PT30M
+  TorcheServeSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Enable access for ssh, inference and management endpoints
+      VpcId: !Ref 'VpcId'
```

Comment:
@maaquib If precreated VPC is used, and it already has these ports being used, will any error get thrown for the users? It will be good to describe as part of readme / troubleshooting steps

Reply:
Resolving this, since a new VPC is now created with the template